host.py

```diff
@@ -305,8 +305,8 @@ def run(
         env_extra: Optional[Dict[str, str]] = None,
         cwd: Optional[str] = None,
     ) -> Result:
-        if self.sudo_needed:
-            cmd = "sudo " + cmd
+        if self.sudo_needed and env is not None:
+            raise ValueError("Setting the full environment with sudo is not supported. Use env_extra instead")
```

Comment:
does it make sense though? `sudo` prepares a clean environment. Setting things like PWD or PATH wouldn't make sense. What exactly do you want to set there?

With `env`, it's not clear which variables the caller really wants to set/delete, and which they just keep there because they don't want to touch them. With sudo (and `_run_remote()`) we run in a different environment, and we need to know which variables should be set/unset and which left alone.

Reply:
I also still miss the reason why we want env_extra

Reply:
Two reasons:

1) https://github.com/bn222/cluster-deployment-automation/pull/133#discussion_r1557565349
2) `env_extra` is more convenient to use than `env` (in my opinion), because it doesn't require you the extra step to copy the current environment (which you almost always want anyway).